tuxemon/client.py

```diff
@@ -134,12 +136,16 @@ def __init__(self, config: TuxemonConfig) -> None:
             "previoussong": None,
         }
 
-        # Set up the command line. This provides a full python shell for
-        # troubleshooting. You can view and manipulate any variables in
-        # the game.
-        self.exit = False  # Allow exit from the CLI
         if self.config.cli:
-            self.cli = cli.CommandLine(self)
+            # TODO: There is no protection for the main thread from the cli
+            # actions that execute in this thread may have undefined
+            # behavior for the game.  at some point, a lock should be
+            # implemented so that actions executed here have exclusive
+            # control of the game loop and state.
+            self.cli = CommandProcessor(local_session)
+            thread = Thread(target=self.cli.run)
```

Comment:
We are playing with fire (although the GIL will prevent memory corruption, and most actions are unlikely to run while state is being changed: humans are not fast). How we did that before without creating a thread?

Reply:
this is how it was run before.  so this isn't a change, im just documenting the problem